src/browser/pages/vscode.ts

```diff
@@ -5,18 +5,65 @@ const options = getOptions()
 // TODO: Add proper types.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-let nlsConfig: any
+// NOTE@jsjoeio
+// This lives here ../../../lib/vscode/src/vs/base/common/platform.ts#L106
+export const nlsConfigElementId = "vscode-remote-nls-configuration"
+
+type NlsConfiguration = {
+  locale: string
+  availableLanguages: { [key: string]: string } | {}
+  _languagePackId?: string
+  _translationsConfigFile?: string
+  _cacheRoot?: string
+  _resolvedLanguagePackCoreLocation?: string
+  _corruptedFile?: string
+  _languagePackSupport?: boolean
+  loadBundle?: any
+}
+
+/**
+ * A helper function to get the NLS Configuration settings.
+ *
+ * This is used by VSCode for localizations (i.e. changing
+ * the display language).
+ *
+ * Make sure to wrap this in a try/catch block when you call it.
+ **/
+export function getNlsConfiguration(document: Document) {
+  const errorMsgPrefix = "[vscode]"
+  const nlsConfigElement = document?.getElementById(nlsConfigElementId)
+  const nlsConfig = nlsConfigElement?.getAttribute("data-settings")
+
+  if (!document) {
+    throw new Error(`${errorMsgPrefix} Could not parse NLS configuration. document is undefined.`)
+  }
+
+  if (!nlsConfigElement) {
+    throw new Error(
+      `${errorMsgPrefix} Could not parse NLS configuration. Could not find nlsConfigElement with id: ${nlsConfigElementId}`,
+    )
+  }
+
+  if (!nlsConfig) {
+    throw new Error(
+      `${errorMsgPrefix} Could not parse NLS configuration. Found nlsConfigElement but missing data-settings attribute.`,
+    )
+  }
+
+  return JSON.parse(nlsConfig) as NlsConfiguration
+}
+
 try {
-  nlsConfig = JSON.parse(document.getElementById("vscode-remote-nls-configuration")!.getAttribute("data-settings")!)
-  if (nlsConfig._resolvedLanguagePackCoreLocation) {
+  const nlsConfig = getNlsConfiguration(document)
+  if (nlsConfig?._resolvedLanguagePackCoreLocation) {
```

Comment:
I think it should no longer be possible for `nlsConfig` to be undefined?

Reply:
Oh right! because now, it can never be `undefined`. I didn't even think to check that again - nice catch!